skimage/_shared/utils.py

```diff
@@ -19,6 +20,54 @@ class skimage_deprecation(Warning):
     pass
 
 
+class change_default_value:
+    """Decorator for changing the default value of an argument.
+
+    Parameters
+    ----------
+    arg_name: str
+        The name of the argument to be updated.
+    old_value: any
+        The argument old value.
+    warning_msg: str
+        Optional warning message. If None, a generic warning message
+        is used.
+    changed_version : str
+        The package version in which the change have been introduced.
+
+    """
+
+    def __init__(self, arg_name, *, new_value, changed_version,
+                 warning_msg=None):
+        self.arg_name = arg_name
+        self.new_value = new_value
+        self.warning_msg = warning_msg
+        self.changed_version = changed_version
+
+    def __call__(self, func):
+        parameters = inspect.signature(func).parameters
+        arg_idx = list(parameters.keys()).index(self.arg_name)
+        old_value = parameters[self.arg_name].default
+
+        if self.warning_msg is None:
+            self.warning_msg = (
+                f"New recommanded value for {self.arg_name} is "
+                f"{self.new_value}. Until version {self.changed_version}, "
+                f"default {self.arg_name} value is {old_value}. Starting "
+                f"from version {self.changed_version}, {self.arg_name} "
+                f"default value will be set to {self.new_value}. To avoid "
+                f"this warning, please explicitly set {self.arg_name} value.")
```

Comment:
```suggestion
                f"The new recommended value for {self.arg_name} is "
                f"{self.new_value}. Until version {self.changed_version}, "
                f"the default {self.arg_name} value is {old_value}. "
                f"From version {self.changed_version}, the {self.arg_name} "
                f"default value will be {self.new_value}. To avoid "
                f"this warning, please explicitly set {self.arg_name} value.")
```